pkg/data/output.go

```diff
@@ -60,16 +72,58 @@ func OutputAsTable(sess *session.Session, outputStream *os.File) error {
 				break
 			}
 		}
-
 		if !isFound {
 			var row []string
 			row = append(row, instance.InstanceType, notApplicable, notApplicable, notApplicable, notApplicable, notApplicable, notApplicable, notApplicable)
 			tableData = append(tableData, row)
 		}
 	}
 	cmdutil.RenderTable(tableData, strings.Split(finalOutputTableHeader, ","), outputStream)
+	fmt.Fprintf(outputStream, "\nDetailed test results can be found in s3://%s/%s\n", testFixture.BucketName, testFixture.BucketRootDir)
+	return nil
+}
 
-	fmt.Fprintf(outputStream, "\nDetailed test results can be found in s3://%s/%s\n", testFixture.BucketName(), testFixture.BucketRootDir())
+// updateResults updates the FinalResult of a test fixture with corresponding CloudWatch data
+// then saves updates results file locally and remotely
+func updateResults(results []*cloudwatch.MetricDataResult, testFixture config.TestFixture) ([]resources.Instance, error) {
+	cwMetrics := make(map[string][]resources.Metric)
+	for _, metricData := range results {
+		fmt.Println(*metricData.Id)
+		if metricData.Values != nil {
+			splitLabel := strings.Split(*metricData.Label, " ")
+			var instanceId string
+			for i, tag := range splitLabel {
+				matched, err := regexp.MatchString(instanceIdRegex, tag)
+				if err != nil {
+					fmt.Println("Could not extract instanceId from MetricDataResult")
+					return nil, err
+				}
+				if matched {
+					instanceId = splitLabel[i]
```

Comment:
it's probably a good idea to check that `instanceId` was set so that you're not adding cwMetrics for the empt string.  Also, once it matches, can you break from the loop? It looks like you're only trying to find instanceId in the tags.